### PR TITLE
[MRG] Regression label for 2d classification data generation

### PR DESCRIPTION
**Pull request by BuenoRuben**
In this branch, I added the 'regression' label to both `_generate_data_2d_classif` and `_generate_data_2d_classif_subspace` in skada/datasets/_sample_generator.py, added a new example for this label: examples/datasets/plot_shifted_dataset_regression.py
and added a new test: `test_make_shifted_datasets_regression `in skada\datasets\tests\test_samples_generator.py

### CODE REVIEW

**Comment by BuenoRuben** [comment]
In this branch, I added the 'regression' label to both `_generate_data_2d_classif` and `_generate_data_2d_classif_subspace` in skada/datasets/_sample_generator.py, added a new example for this label: examples/datasets/plot_shifted_dataset_regression.py
and added a new test: `test_make_shifted_datasets_regression `in skada\datasets\tests\test_samples_generator.py

File: examples/datasets/plot_shifted_dataset_regression.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    fig.suptitle(shift.replace("_", " ").title(), fontsize=14)
+    plt.subplots_adjust(bottom=0.15)
+    s = ax1.scatter(
+        X_source[:, 0],
+        X_source[:, 1],
+        c=y_source*10,
+        vmax=1,
+        alpha=0.5,
+    )
+    cb=fig.colorbar(s)
```
Comment (tgnassou) [inline examples/datasets/plot_shifted_dataset_regression.py:59]: I think one colorbar should be enough per line

File: skada/datasets/tests/test_samples_generator.py [hunk 2]
```diff
@@ -150,6 +150,35 @@ def test_make_multi_source_shifted_datasets(shift):
...
+    X_source, y_source, X_target, y_target = check_X_y_domain(
+        X,
+        y=y,
+        sample_domain=sample_domain,
+        return_joint=False,
+    )
+
+    assert X_source.shape == (10 * 8, 2), "X source shape mismatch"
+    assert y_source.shape == (10 * 8,), "y source shape mismatch"
+    assert (max(y_source) <=1) and min(y_source) >=0, "Wrong y-values in source domain, probabilities should be between 0 and 1"
```
Comment (tgnassou) [inline skada/datasets/tests/test_samples_generator.py:174]: What is this test doing? y is a value, not a probability?

**Review by tgnassou — COMMENTED** [review]
I think in the code I gave you, there was also a regression mode for subspace dataset, it could be nice to have it too

**Comment by tgnassou** [comment]
Your tests are not passing

File: examples/datasets/plot_shifted_dataset_regression.py (hunk 1 above)
Comment (BuenoRuben) [reply examples/datasets/plot_shifted_dataset_regression.py]: it's done

**Pull request merged**